sympy/physics/vector/vector.py

```diff
@@ -453,6 +453,29 @@ def _det(mat):
     __rand__ = __and__
     __rmul__ = __mul__
 
+    def components(self):
+        """
+        Returns the components of this vector in the form of a
+        Python dictionary mapping BaseVector instances to the
+        corresponding measure numbers.
+        Examples
+        ========
+        >>> from sympy.physics.vector import ReferenceFrame
+        >>> R = ReferenceFrame('R')
+        >>> v = 3*R.x + 4*R.y + 5*R.z
+        >>> v.components()
+        {R.x: 3, R.y: 4, R.z: 5}
+        >>> v = 3*R.x + 4*R.y
+        >>> v.components()
+        {R.x: 3, R.y: 4, R.z: 0}
+        """
+        components = {}
+        components[(self.args[0][1]).x] = self.args[0][0][0]
+        components[(self.args[0][1]).y] = self.args[0][0][1]
+        components[(self.args[0][1]).z] = self.args[0][0][2]
+
+        return components
```

Comment:
There is already something like this, `.seperate()` maybe?

Also, probably best to make this private for now.

Reply:
> There is already something like this, `.seperate()` maybe?

I went through `.separate()` method , it is used when 2 vectors of different reference frames are added and gives
constituents of the vector in different reference frames.

Reply:
Ok, probably just keep it private with a leading underscore we it is only used internally. We don't need to expose these things to the users (same goes for `seperate()` but that made it in as public is seems).